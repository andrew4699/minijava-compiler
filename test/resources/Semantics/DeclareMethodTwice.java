class Factorial{
	public static void main(String[] a){
		System.out.println(new A().ComputeFac(10));
	}
}

class A {
	int a;
	boolean b;

	public int ComputeFac(int num){
		int num_aux ;
		if (num < 1)
				num_aux = 1 ;
		else 
				num_aux = num * (this.ComputeFac(num-1)) ;
		return num_aux ;
	}

  public int ComputeFac(int num1point5) {
    return 2;
  }
}

class B extends A {
  public int ComputeFac(int num2) {
    return num2;
  }
}

class C extends B {
  public int ComputeFac(int num3) {
    return 10;
  }

  public int ComputeFac(int num4) {
    return 15;
  }
}
