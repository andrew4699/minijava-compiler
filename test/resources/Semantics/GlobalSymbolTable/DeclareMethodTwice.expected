[GlobalSymbolTableVisitor - 20]: Method ComputeFac already declared
[GlobalSymbolTableVisitor - 36]: Method ComputeFac already declared
Global Symbol Table:

Class: Factorial
  Superclass: None

  Fields:

  Methods:
    void main(String[] a)

Class: A
  Superclass: None

  Fields:
    int a
    boolean b

  Methods:
    int ComputeFac(int num)

Class: B
  Superclass: A

  Fields:

  Methods:
    int ComputeFac(int num2)

Class: C
  Superclass: B

  Fields:

  Methods:
    int ComputeFac(int num3)
